dev-python/cadquery-ocp/cadquery-ocp-7.5.2_rc20210706.ebuild

```
@@ -0,0 +1,160 @@
+
+	# Absolute dirname of OCCT's include and shared library directories.
+	local _OCCT_INCLUDE_DIR="${EPREFIX}/usr/include/opencascade-${OCCT_PV}"
+	local _OCCT_LIB_DIR="${EPREFIX}/usr/lib64/opencascade-${OCCT_PV}"
+
+	# Absolute dirname of VTK's include directory,
+	local _VTK_INCLUDE_DIR="${EPREFIX}/usr/include/vtk-${_VTK_VERSION}"
+
+	# Ensure the above paths exist as a crude sanity test.
+	test -d "${_CLANG_INCLUDE_DIR}" || die "${_CLANG_INCLUDE_DIR} not found."
+	test -f "${_CLANG_LIB_FILE}"    || die "${_CLANG_LIB_FILE} not found."
+	test -d "${_OCCT_INCLUDE_DIR}"  || die "${_OCCT_INCLUDE_DIR} not found."
+	test -d "${_OCCT_LIB_DIR}"      || die "${_OCCT_LIB_DIR} not found."
+	test -d "${_VTK_INCLUDE_DIR}"   || die "${_VTK_INCLUDE_DIR} not found."
+
+	# "dev-python/clang-python" atom targeting this Clang version.
+	local _CLANG_PYTHON_ATOM="dev-python/clang-python-${_CLANG_VERSION}"
+
+	# Ensure "dev-python/clang-python" targets this Clang version.
+	has_version -r "=${_CLANG_PYTHON_ATOM}" ||
+		die "${_CLANG_PYTHON_ATOM} not installed."
+
+	# Remove all vendored subdirectories.
+	rm -rf conda opencascade pywrap || die
+
+	# Inject a symlink to OCCT's include directory.
+	ln -s "${_OCCT_INCLUDE_DIR}" opencascade || die
+
+	python_copy_sources
+	cadquery-ocp_src_prepare() {
+		cd "${BUILD_DIR}" || die
+
+		# Generate OCCT bindings in the "${BUILD_DIR}/OCP/" subdirectory.
+		${EPYTHON} -m bindgen \
+			--verbose \
+			--njobs $(get_nproc) \
+			--libclang "${_CLANG_LIB_FILE}" \
+			--include "${_CLANG_INCLUDE_DIR}" \
+			--include "${_VTK_INCLUDE_DIR}" \
+			all ocp.toml || die
+
+		# Remove the source "FindOpenCascade.cmake" after generating bindings,
+		# which copied that file to the target "OCP/FindOpenCascade.cmake".
+		rm FindOpenCascade.cmake || die
+
+		#FIXME: Submit an upstream issue. This is frankly awful.
+		# Replace all hardcoded paths in "OCP/FindOpenCascade.cmake" with
+		# standard OCCT paths derived above. That file is both fundamentally
+		# broken and useless, as the ${CASROOT} environment variable and
+		# "/usr/lib64/cmake/opencascade-${PV}/OpenCASCADEConfig.cmake" file
+		# already reliably identify all requisite OpenCASCADE paths. Failure to
+		# patch this file results in src_configure() failures resembling:
+		#     -- Could NOT find OPENCASCADE (missing: OPENCASCADE_LIBRARIES)
+		sed -i \
+			-e 's~$ENV{CONDA_PREFIX}/include/opencascade\b~'${_OCCT_INCLUDE_DIR}'~' \
+			-e 's~$ENV{CONDA_PREFIX}/lib\b~'${_OCCT_LIB_DIR}'~' \
+			-e 's~$ENV{CONDA_PREFIX}/Library/\(lib\|include/opencascade\)~~' \
+			OCP/FindOpenCascade.cmake || die
+	}
+	python_foreach_impl cadquery-ocp_src_prepare
+}
+
+src_configure() {
+	cadquery-ocp_src_configure() {
+		cd "${BUILD_DIR}" || die
+		cmake \
+			-Wno-dev \
+			-S OCP/ \
+			-B OCP.binary_tree/ \
+			-G Ninja \
+			-D CMAKE_BUILD_TYPE=Gentoo \
+			-D PYTHON_EXECUTABLE="${PYTHON}" \
+			|| die
+	}
+	python_foreach_impl cadquery-ocp_src_configure
```

```suggestion
	python_foreach_impl CMAKE_USE_DIR="${BUILD_DIR}/OCP" cmake_src_configure
```
(and maybe something with `my_cmake_args`)